show/main.py

```diff
@@ -768,23 +771,241 @@ def get_bgp_peer():
+                        str_2_print += " [" + str(info[i]['distance']) + "/" + str(info[i]['metric']) + "]"
+                    str_length = len(str_2_print)
+                else:
+                    str_2_print += " "*(str_length - 3)
+                if "ip" in info[i]['nexthops'][j]:
+                    str_2_print += " via " + info[i]['nexthops'][j]['ip'] + ","
+                    if "interfaceName" in info[i]['nexthops'][j]:
+                        str_2_print += " " + info[i]['nexthops'][j]['interfaceName'] + ","
+                elif "directlyConnected" in info[i]['nexthops'][j]:
+                    str_2_print += " is directly connected,"
+                    if "interfaceName" in info[i]['nexthops'][j]:
+                        str_2_print += " " + info[i]['nexthops'][j]['interfaceName'] + ","
+                elif "unreachable" in info[i]['nexthops'][j]:
+                    if "reject" in info[i]['nexthops'][j]:
+                        str_2_print += " (ICMP unreachable)"
+                    elif "admin-prohibited" in info[i]['nexthops'][j]:
+                        str_2_print += " (ICMP admin-prohibited)"
+                    elif "blackhole" in info[i]['nexthops'][j]:
+                        str_2_print += " (blackhole)"
+                if "vrf" in info[i]['nexthops'][j]:
+                    str_2_print += "(vrf " + info[i]['nexthops'][j]['vrf'] + ", " + info[i]['nexthops'][j]['interfaceName'] + ","
+                if "active" not in info[i]['nexthops'][j]:
+                    str_2_print += " inactive"
+                if "onLink" in info[i]['nexthops'][j]:
+                    str_2_print += " onlink"
+                if "recursive" in info[i]['nexthops'][j]:
+                    str_2_print += " (recursive)"
+                if "source" in info[i]['nexthops'][j]:
+                    str_2_print += ", src " + info[i]['nexthops'][j]['source']
+                if "labels" in info[i]['nexthops'][j]:
+                    # MPLS labels are stored as an array (list) in json if present. Need to print through each one in list
+                    str_2_print += ", label "
+                    for k in range(0, len(info[i]['nexthops'][j]['labels'])):
+                        # MPLS labels that has value 15 or lower has special interpretation
+                        if info[i]['nexthops'][j]['labels'][k] > 15:
+                            label_string = str(info[i]['nexthops'][j]['labels'][k])
+                        else:
+                            label_string = mpls_label_code[info[i]['nexthops'][j]['labels'][k]]
+                        if k == 0:
+                            str_2_print += label_string
+                        else:
+                            str_2_print += "/" + label_string
+                # add uptime at the end of the string
+                str_2_print += " " + info[i]['uptime']
+                # print out this string
+                print str_2_print
```

Comment:
This too long, lets split it up. There are few types of next hops. Each type can have it's own render function that returns a formatted string. Also these functions can be in utils.py or you can also create a route.py. 
show/main.py is the main CLI, this can just call an API that returns a formatted string.

Wondering for formatting if we can use python string formats.

Reply:
Sure. Let me split up at section where it is looking at the same field but with different possibilities.
Unfortunately this logic here is not something sharable to other purpose so will not be moving this to other files...

Reply:
if you use 'tabulate' package python will do the formatting for you.